CMakeLists.txt

```diff
@@ -1174,6 +1174,25 @@ if(HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION)
   hpx_add_config_define(HPX_HAVE_CXX14_RETURN_TYPE_DEDUCTION)
 endif()
 
+################################################################################
+# Colored output (when using ninja)
+################################################################################
+if (CMAKE_MAKE_PROGRAM MATCHES ninja)
+  hpx_option(CMAKE_FORCE_COLORED_OUTPUT BOOL
+      "Always produce ANSI-colored output (GNU/Clang only)." TRUE ADVANCED)
+else ()
+  hpx_option(CMAKE_FORCE_COLORED_OUTPUT BOOL
+    "Always produce ANSI-colored output (GNU/Clang only)." FALSE ADVANCED)
+endif ()
+
+if (${CMAKE_FORCE_COLORED_OUTPUT})
+  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    add_compile_options (-fdiagnostics-color=always)
```

Comment:
Can you change this to `hpx_add_compile_option` (I think)? We wouldn't want this to leak this to someone who uses the `hpx` target (once that's in place).